src/Components/FakeTargetsOutline.fs

```diff
@@ -0,0 +1,366 @@
+            refresh.fire textEditor.document.uri
+
+    let private onDidChangeActiveTextEditor (textEditor : TextEditor) =
+        setShowTargetsOutlineForEditor textEditor
+        if textEditor = undefined || (not (isFsharpFile textEditor.document)) then
+            reallyRefresh.fire(None)
+
+
+    type [<Pojo>] RequestLaunch =
+        { name : string
+          ``type`` : string
+          request : string
+          preLaunchTask : string option
+          program : string
+          args : string array
+          cwd : string
+          console : string
+          stopAtEntry : bool 
+          justMyCode : bool
+          requireExactSource : bool }
+
+    let activate (context : ExtensionContext) =
+        setShowTargetsOutlineForEditor window.activeTextEditor
+
+        let treeViewId = ShowInActivity.initializeAndGetId ()
+
+        window.onDidChangeActiveTextEditor.Invoke(unbox onDidChangeActiveTextEditor)
+            |> context.subscriptions.Add
+
+        refresh.event.Invoke(fun uri ->
+            if isEnabledFor uri then
+                reallyRefresh.fire(None)
+            createEmpty)
+            |> context.subscriptions.Add
+
+        workspace.onDidChangeConfiguration.Invoke(unbox onDidChangeConfiguration)
+            |> context.subscriptions.Add
+
+        commands.registerCommand("FAKE.targetsOutline.goTo", Func<obj, obj>(fun n ->
+            let line =
+                let m = unbox<Model> n
+                m.Declaration.Line
+
+            let args =
+                createObj [
+                    "lineNumber" ==> line - 1
+                    "at" ==> "center"
+                ]
+
+            vscode.commands.executeCommand("revealLine", args)
+            |> unbox
+        )) |> context.subscriptions.Add
+
+        let runFake doDebug onlySingleTarget targetName =
+            promise {
+                match currentDocument with
+                | Some scriptFile ->
+                    let scriptDir = node.path.dirname scriptFile
+                    let scriptName = node.path.basename scriptFile
+                    let! fakeRuntime = LanguageService.FakeSupport.fakeRuntime()
+                    let preArg = if onlySingleTarget then "-st" else "-t"
+                    let args =
+                        if doDebug then [| "run"; "--fsiargs"; "--debug:portable --optimize-"; scriptName; preArg; targetName |]
+                        else [| "run"; scriptName; preArg; targetName |]
+                    let cfg : RequestLaunch = 
+                        { name = "Fake Script Debugging"
+                          ``type`` = "coreclr"
+                          request = "launch"
+                          preLaunchTask = None
+                          program = fakeRuntime
+                          args = args
+                          cwd = scriptDir
+                          console = "externalTerminal"
+                          stopAtEntry = false
+                          justMyCode = false
```

Comment:
I'm not sure why we need to deactivate this or what "my Code" means here. If I set this to `true` no breakpoint hits.
How can I tell the debugger that the current file is "my code"? Anyone understands the details here?
The problem with `false` is that you can see all the stack from the FAKE runtime and potentially unrelated stuff...

Reply:
Can someone please ping someone who knows something?

Reply:
I believe this is exactly the same as Just My Code in VS - https://docs.microsoft.com/en-us/visualstudio/debugger/just-my-code?view=vs-2019

Reply:
> The .NET Framework debugger considers optimized binaries and non-loaded .pdb files to be non-user code.
> Three compiler attributes also affect what the .NET debugger considers to be user code:
>
>    DebuggerNonUserCodeAttribute tells the debugger that the code it's applied to isn't user code.
>    DebuggerHiddenAttribute hides the code from the debugger, even if Just My Code is turned off.
>     DebuggerStepThroughAttribute tells the debugger to step through the code it's applied to, rather than step into the code.
>
> The .NET Framework debugger considers all other code to be user code.

Ok none of these attributes was set but I saw in the debug output:

`Not loading <compiledDll_hash>.pdb because just my code is enabled`.
So the debugger is not loading the pdb because just my code is enabled and it considers the dll not 'my code' because no pdb is loaded? Circle reasoning? I'm sure the dll has `--optimize-` enabled

Reply:
Should we ignore this for now and note as "known limitation"? To be honest, I have no idea what to do. 
Just unfortunate that the experience is a bit unfortunate... At least it works.